admin/lib/validator/check-window-v2/check-window-add-validator.js

```diff
@@ -0,0 +1,146 @@
+    checkWindowData['familiarisationCheckStartMonth'], checkWindowData['familiarisationCheckStartYear'])
+  const familiarisationCheckEndDate = dateService.createUTCFromDayMonthYear(checkWindowData['familiarisationCheckEndDay'],
+    checkWindowData['familiarisationCheckEndMonth'], checkWindowData['familiarisationCheckEndYear'])
+  const liveCheckStartDate = dateService.createUTCFromDayMonthYear(checkWindowData['liveCheckStartDay'],
+    checkWindowData['liveCheckStartMonth'], checkWindowData['liveCheckStartYear'])
+  const liveCheckEndDate = dateService.createUTCFromDayMonthYear(checkWindowData['liveCheckEndDay'],
+    checkWindowData['liveCheckEndMonth'], checkWindowData['liveCheckEndYear'])
+
+  // Compare date fields
+  // Admin start date
+  if (adminStartDate && adminStartDate.isAfter(familiarisationCheckStartDate)) {
+    validationError.addError('adminStartDateAfterFamiliarisationCheckStartDate', true)
+  }
+  if (adminStartDate && adminStartDate.isAfter(liveCheckStartDate)) {
+    validationError.addError('adminStartDateAfterLiveCheckStartDate', true)
+  }
+  // Admin end date
+  if (adminEndDate && adminEndDate.isBefore(adminStartDate)) {
+    validationError.addError('adminEndDateBeforeAdminStartDate', true)
+  }
+  if (adminEndDate && adminEndDate.isBefore(liveCheckEndDate)) {
+    validationError.addError('adminEndDateBeforeLiveCheckEndDate', true)
+  }
+  if (adminEndDate && adminEndDate.isBefore(familiarisationCheckEndDate)) {
+    validationError.addError('adminEndDateBeforeFamiliarisationCheckEndDate', true)
+  }
+  // Familiarisation check start date
+  if (familiarisationCheckStartDate && familiarisationCheckStartDate.isAfter(liveCheckStartDate)) {
+    validationError.addError('familiarisationCheckStartDateAfterLiveCheckStartDate', true)
+  }
+  if (familiarisationCheckStartDate && familiarisationCheckStartDate.isAfter(familiarisationCheckEndDate)) {
+    validationError.addError('familiarisationCheckStartDateAfterFamiliarisationCheckEndDate', true)
+  }
+  // Familiarisation check end date
+  if (familiarisationCheckEndDate && familiarisationCheckEndDate.isBefore(adminStartDate)) {
+    validationError.addError('familiarisationCheckEndDateBeforeAdminStartDate', true)
+  }
+  // Live check start date
+  if (liveCheckStartDate && liveCheckStartDate.isAfter(liveCheckEndDate)) {
+    validationError.addError('liveCheckStartDateAfterLiveCheckEndDate', true)
+  }
+  // Live check end date
+  if (liveCheckEndDate && liveCheckEndDate.isBefore(adminStartDate)) {
+    validationError.addError('liveCheckEndDateBeforeAdminStartDate', true)
+  }
+  if (liveCheckEndDate && familiarisationCheckEndDate && !familiarisationCheckEndDate.isSame(liveCheckEndDate)) {
+    validationError.addError('familiarisationCheckEndDateNotEqualLiveCheckEndDate', true)
+  }
+  return validationError
+}
+
+checkWindowAddValidator.generateValidationData = (dateData) => {
```

Comment:
Please don't do this!  No function should have 13 arguments.  Consider using a fluent interface instead.

Reply:
@jon-shipley this is now updated.